index.html

```diff
@@ -135,8 +135,23 @@
 
   <section id="discovery">
     <h2>Discovery</h2>
-    <p>A potential subscriber initiates discovery by retrieving (GET or HEAD request) the topic to which it wants to subscribe. The HTTP [[RFC7231]] response from the publisher MUST include at least one Link Header [[!RFC5988]] with <samp>rel=hub</samp> (a hub link header) as well as exactly one Link Header [[!RFC5988]] with <samp>rel=self</samp> (the self link header). The former MUST indicate the exact URL of a PubSub hub designated by the publisher. If more than one URL is specified, it is expected that the publisher pings each of these URLs, so the subscriber may subscribe to one or more of these. The latter will point to the permanent URL for the resource for which notifications will be sent.</p>
-    <p>In the absence of HTTP [[!RFC7231]] Link headers, subscribers MAY fall back to other methods to discover the hub(s) and the canonical URI of the topic. If the topic is an XML based feed, it MAY use embedded link elements as described in Appendix B of Web Linking [[!RFC5988]]. Similarly, for HTML pages, it MAY use embedded link elements as described in Appendix A of Web Linking [[!RFC5988]]. Finally, publishers MAY also use the Well-Known Uniform Resource Identifiers [[!RFC5785]] .host-meta to include the &lt;Link&gt; element with rel="hub".</p>
+    <p>The discovery mechanism aims at identifying at least 2 URLs.</p>
+    <ul>
+      <li>The exact URL of the hub(s) designated by the publisher. If more than one URL is specified, it is expected that the publisher pings each of these URLs, so the subscriber may subscribe to one or more of these.</li>
+      <li>The canonical URL for the topic to which subscribers are expected to use for subscriptions.</li>
+    </ul>
+    <p>The protocol currently supports the following discovery mechanisms. Publishers MUST implement at least one of them:</p>
+    <ul>
+      <li>Link Headers [[!RFC5988]]: the publisher SHOULD include at least one Link Header [[!RFC5988]] with <samp>rel=hub</samp> (a hub link header) as well as exactly one Link Header [[!RFC5988]] with <samp>rel=self</samp> (the self link header)</li>
+      <li>If the topic is an XML based feed, publishers SHOULD use embedded link elements as described in Appendix B of Web Linking [[!RFC5988]]. Similarly, for HTML pages, publishers SHOULD use embedded link elements as described in Appendix A of Web Linking [[!RFC5988]]</li>
+      <li class="warning">Finally, publishers MAY also use the Well-Known Uniform Resource Identifiers [[!RFC5785]] <samp>.host-meta</samp> to include the &lt;Link&gt; element with rel="hub". However, please note that this mechanism is currently at risk and may be deprecated. </li>
+    </ul>
+    <p>When perfoming discovery, subscribers MUST implement all three discovery mechanisms but should respect the following order.</p>
```

Comment:
sounds good, but I would recommend to also fix the host-meta wording before merging.